coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -724,41 +725,37 @@ private static Schema setupNestedNamespace(@Nonnull Schema schema, @Nonnull Stri
         return schema;
       case MAP:
         Schema valueSchema = schema.getValueType();
-        Schema valueSchemaWithNestedNamespace = setupNestedNamespace(valueSchema, namespace);
-        Schema mapSchema = Schema.createMap(valueSchemaWithNestedNamespace);
-
-        return mapSchema;
+        Schema valueSchemaWithNestedNamespace = setupNestedNamespace(valueSchema, namespace, defaultValue);
+        return Schema.createMap(valueSchemaWithNestedNamespace);
       case ARRAY:
         Schema elementSchema = schema.getElementType();
-        Schema elementSchemaWithNestedNamespace = setupNestedNamespace(elementSchema, namespace);
-        Schema arraySchema = Schema.createArray(elementSchemaWithNestedNamespace);
-
-        return arraySchema;
+        Schema elementSchemaWithNestedNamespace = setupNestedNamespace(elementSchema, namespace, defaultValue);
+        return Schema.createArray(elementSchemaWithNestedNamespace);
       case ENUM:
-        Schema enumSchemaWithNestedNamespace =
-            Schema.createEnum(schema.getName(), schema.getDoc(), namespace, schema.getEnumSymbols());
-
-        return enumSchemaWithNestedNamespace;
+        return Schema.createEnum(schema.getName(), schema.getDoc(), namespace, schema.getEnumSymbols());
       case RECORD:
-        Schema recordSchema = setupNestedNamespaceForRecord(schema, namespace);
-        return recordSchema;
+        return setupNestedNamespaceForRecord(schema, namespace);
       case UNION:
         List<Schema> types = new ArrayList<>();
         if (isNullableType(schema)) {
```

Comment:
Does `schema` have the correct ordering? Can we preserve it regardless of what the default value is?

Reply:
schema will have the same ordering as the underlying table/view. It may or may not be correct. 
We can simply preserve it I think. But In that implementation choice, we won't be able to 'fix' it for corrupted tables.

Reply:
I would preserve it. It is good to flag corrupted tables.

Reply:
I have preserved the underlying ordering for union type irrespective of the nullability/default value check. 
Now, this has led to another change. When we generate a new field on the fly, for e.g., apply a transformation like count() / max() / unnest() on a nullable column, the schema of this new field appears to be `["long", "null"]` instead of `["null", "long"]` as before. Technically, both representations are correct. But we would be changing the semantics. 

Is it possible that it might lead to regression for some usecases?

Reply:
Thanks @aastha25! As long as these are not assigned a default value (which should be the case since they are derived fields), these options should be equivalent.

Reply:
@funcheetah, a large number of test cases has been impacted by this. Could you take a look?

Reply:
Yeah, I agree that if there is no default value set, they should be equivalent. Though a little against Avro optional fields recommendation (null as the first option).

Reply:
So we can either go with the simplified implementation. In this case, we will always generate schema `["long", "null"]` for derived optional columns. It is technically correct but not recommended.

Or we can go with the earlier implementation, where we take into consideration the `defaultValue` and return the recommmend schema type. 

I'm fine with either. If we choose the simplified implementation, this issue might come back to us again? If Avro makes their recommendation mandatory at some point or if some user has an issue due to the inversion?

Reply:
@aastha25, for backward compatibility, are we able to maintain the nulls first for derived fields? Which part is responsible for generating `["long", "null"]` (and not the opposite) in the first place?

Reply:
When traversing the RelNodePlan, we land upon `visit(LogicalAggregate logicalAggregate)` 

and then this line https://github.com/linkedin/coral/blob/b2ef3ef62de1d6769a6de0183526ea531f2ce16e/coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java#L217 
here, you can see we are appending null later.

Reply:
I do not recall specific reason. Actually, I think it is better to put null as first option. We can use integration tests to catch potential regressions after this change.

Reply:
+1. That sounds like the best option here.

Reply:
Tried out this change and it seems to be backward compatible now.